src/test/java/seedu/address/testutil/TypicalVisits.java

```diff
@@ -37,8 +58,55 @@ public static VisitBook getTypicalVisitBook() {
         return vb;
     }
 
+    /**
+     * Returns a {@code VisitBook} with all the all visits.
+     */
+    public static VisitBook getWholeVisitBook() {
+        VisitBook vb = new VisitBook();
+        for (Visit visit : getAllVisits()) {
+            vb.addVisit(visit);
+        }
+        return vb;
+    }
+
+    /**
+     * Returns a {@code VisitBook} with more than 60% visits are infected visits.
+     */
+    public static VisitBook getMoreThanSixtyPercentVisitBook() {
+        VisitBook vb = new VisitBook();
+        for (Visit visit : getMoreThanSixtyPercentInfectedVisits()) {
+            vb.addVisit(visit);
+        }
+        return vb;
+    }
+
     public static List<Visit> getTypicalVisits() {
         return new ArrayList<>(Arrays.asList(FIRST_VISIT, SECOND_VISIT, THIRD_VISIT,
                 FOURTH_VISIT, FIFTH_VISIT, SIXTH_VISIT));
     }
+
+    public static List<Index> getLocationIdsOfTypicalVisits() {
+        return new ArrayList<>(Arrays.asList(INDEX_SECOND, INDEX_THIRD, INDEX_FIRST, INDEX_FOURTH,
+                INDEX_FIFTH, INDEX_SECOND));
+    }
+
+    public static List<Visit> getAllVisits() {
+        return new ArrayList<>(Arrays.asList(FIRST_VISIT, SECOND_VISIT, THIRD_VISIT,
+                FOURTH_VISIT, FIFTH_VISIT, SIXTH_VISIT, SEVENTH_VISIT, EIGHTH_VISIT,
+                NINTH_VISIT, TENTH_VISIT, ELEVENTH_VISIT, TWELFTH_VISIT));
+    }
+
+    public static List<Visit> getVisitsForTest() {
+        return new ArrayList<>(Arrays.asList(FIRST_VISIT, SECOND_VISIT, FOURTH_VISIT, FIFTH_VISIT,
+                SIXTH_VISIT, SEVENTH_VISIT, EIGHTH_VISIT, NINTH_VISIT, TENTH_VISIT, ELEVENTH_VISIT));
+    }
+
+    public static List<Index> getLocationsIdsFormVisitsForTest() {
+        return new ArrayList<>(Arrays.asList(INDEX_SECOND, INDEX_THIRD, INDEX_FIFTH, INDEX_FOURTH));
+    }
+
+    public static List<Visit> getMoreThanSixtyPercentInfectedVisits() {
+        return new ArrayList<>(Arrays.asList(FIRST_VISIT, SECOND_VISIT, SEVENTH_VISIT, EIGHTH_VISIT,
+                NINTH_VISIT, TENTH_VISIT, ELEVENTH_VISIT, TWELFTH_VISIT));
```

Comment:
Perhaps some documentation or name changing could be done? Its hard for me to directly understand the significance between 60 percent and the visits chosen.

Reply:
Done